### PR TITLE
Configure custom/insecure registry to search path

### PR DESCRIPTION
**Pull request by josgonza-rh**
Configure custom/insecure registry to search path

### CODE REVIEW

File: content/registries/_index.md [hunk 1]
```diff
@@ -13,3 +13,49 @@ weight: 23
 oc patch image.config.openshift.io/cluster -p \
 '{"spec":{"allowedRegistriesForImport":[{"domainName":"my.own.registry.example.com:8888","insecure":true}],"registrySources":{"insecureRegistries":["my.own.registry.example.com:8888"]}}}' --type='merge'
 ```
+
+# Configure custom/insecure registry to search path
```
Comment (e-minguez) [inline content/registries/_index.md]: Please add `(4.3)` to the header to clarify it is only valid in 4.3

File: content/registries/_index.md [hunk 2]
```diff
@@ -13,3 +13,49 @@ weight: 23
...
+
+There seems to be no way to remove/modify or append any additional entries to unqualified-search-registries line in /etc/containers/registries.conf file as of now (OCP 4.3).
```
Comment (e-minguez) [inline content/registries/_index.md]: Do you have the bugzilla/github issue where this will be fixed? If so, please add it as a reference

File: content/registries/_index.md [hunk 3]
```diff
@@ -13,3 +13,49 @@ weight: 23
...
+
+There is way to modify the file using machineconfig which gets overwritten by editing `image.config.openshift.io/cluster` which will execute `99-master/worker-<uuid>-registries` and bring back the `unqualified-search-registries` again to the file, so this isn't the recommended way to perform the change. So to avoid to modify that CR, the machineconfig do the job that should be done by `image.config.openshift.io/cluster`.
```
Comment (e-minguez) [inline content/registries/_index.md]: I don't understand this :sweat_smile: can you rephrase it please?

File: content/registries/_index.md [hunk 4]
```diff
@@ -13,3 +13,49 @@ weight: 23
 oc patch image.config.openshift.io/cluster -p \
 '{"spec":{"allowedRegistriesForImport":[{"domainName":"my.own.registry.example.com:8888","insecure":true}],"registrySources":{"insecureRegistries":["my.own.registry.example.com:8888"]}}}' --type='merge'
 ```
+
+# Configure custom/insecure registry to search path
+
+There seems to be no way to remove/modify or append any additional entries to unqualified-search-registries line in /etc/containers/registries.conf file as of now (OCP 4.3).
+
+There is way to modify the file using machineconfig which gets overwritten by editing `image.config.openshift.io/cluster` which will execute `99-master/worker-<uuid>-registries` and bring back the `unqualified-search-registries` again to the file, so this isn't the recommended way to perform the change. So to avoid to modify that CR, the machineconfig do the job that should be done by `image.config.openshift.io/cluster`. 
+
```
Comment (e-minguez) [inline content/registries/_index.md]: Is the following object something that needs to be applied? Can you ellaborate on the instructions? Thanks!

File: content/registries/_index.md (hunk 1 above)
Comment (josgonza-rh) [reply content/registries/_index.md]: It is 4.x where x is <= 3

File: content/registries/_index.md (hunk 2 above)
Comment (josgonza-rh) [reply content/registries/_index.md]: At this moment there is no BZ or fix for OCP <= 4.3 (I think is in the roadmap for 4.4-5)

File: content/registries/_index.md (hunk 4 above)
Comment (josgonza-rh) [reply content/registries/_index.md]: Sure !!

You must apply the MachineConfig (master, worker or both it depends of your needs)
```
oc create -f 99_<MachineConfigPool>_container_runtime.yaml
```
Then wait for clusteroperators to finish and nodes to be ready and then one can confirm the modification by executing the command below:
```
oc wait mcp/<MachineConfigPool> --for condition=updated --timeout=600s
```

File: content/registries/_index.md (hunk 3 above)
Comment (josgonza-rh) [reply content/registries/_index.md]: At this moment, there seems to be no way to remove/modify or append any additional entries to unqualified-search-registries line in /etc/containers/registries.conf file as of now (OCP <=4.3).

A workaround to modify the file is using a machineconfig that do the job that should be done by `image.config.openshift.io/cluster` (full control of the /etc/containers/{policy.json,registries.conf} content).

Yo have to be aware that it could gets overwritten by editing `image.config.openshift.io/cluster` which will execute `99-master/worker-<uuid>-registries` and bring back the `unqualified-search-registries` again to the file, so you must avoit to use the `image.config.openshift.io/cluster` afeter you applied the machineconfig.

File: content/registries/_index.md (hunk 1 above)
Comment (e-minguez) [reply content/registries/_index.md]: Could you please add those changes to the PR so I can merge it? Thanks

File: content/registries/_index.md (hunk 2 above)
Comment (e-minguez) [reply content/registries/_index.md]: :+1:

File: content/registries/_index.md (hunk 3 above)
Comment (e-minguez) [reply content/registries/_index.md]: Could you please add those changes to the PR so I can merge it? Thanks

File: content/registries/_index.md (hunk 4 above)
Comment (e-minguez) [reply content/registries/_index.md]: Could you please add those changes to the PR so I can merge it? Thanks

File: content/registries/_index.md [hunk 5]
```diff
@@ -13,3 +13,61 @@ weight: 23
 oc patch image.config.openshift.io/cluster -p \
 '{"spec":{"allowedRegistriesForImport":[{"domainName":"my.own.registry.example.com:8888","insecure":true}],"registrySources":{"insecureRegistries":["my.own.registry.example.com:8888"]}}}' --type='merge'
 ```
+
+# Configure custom/insecure registry to search path in OCP 4.x
+
+At this moment, there seems to be no way to remove/modify or append any additional entries to unqualified-search-registries line in /etc/containers/registries.conf file as of now (OCP <=4.3).
```
Comment (e-minguez) [inline content/registries/_index.md]: What about this?:

In OpenShift 4, the registries configuration is managed by the [Image Registry Operator](https://docs.openshift.com/container-platform/4.3/registry/configuring-registry-operator.html). In order to modify registries parameter, it is only required to modify the `image.config.openshift.io/cluster` object, that manages  the `/etc/containers/{policy.json,registries.conf}` files content under the hood.

In OpenShift versions prior to 4.4 a missing feature of the operator is the ability to remove, modify or append any additional entry to the `unqualified-search-registries` line in the `/etc/containers/registries.conf` file to allow search in insecure registries.

File: content/registries/_index.md [hunk 6]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+
+A workaround to modify the file is using a machineconfig that do the job that should be done by `image.config.openshift.io/cluster` (full control of the /etc/containers/{policy.json,registries.conf} content).
```
Comment (e-minguez) [inline content/registries/_index.md]: Meanwhile, the current workaround is to modify the `/etc/containers/{policy.json,registries.conf}` files using a `machineconfig` object instead.

File: content/registries/_index.md [hunk 7]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+
+Yo have to be aware that it could gets overwritten by editing `image.config.openshift.io/cluster` which will execute `99-master/worker-<uuid>-registries` and bring back the `unqualified-search-registries` again to the file, so you must avoit to use the `image.config.openshift.io/cluster` afeter you applied the machineconfig.
```
Comment (e-minguez) [inline content/registries/_index.md]: You need to be aware that these files will be overwritten if the `image.config.openshift.io/cluster` object is modified, as they are intended to be managed by the operator. If the object is modifed, the operator will modify the `99-master/worker-<uuid>-registries` `machineconfig` object and will bring back the previous `unqualified-search-registries` content to the file, so you must avoid to use the `image.config.openshift.io/cluster` after you applied the `machineconfig`.

File: content/registries/_index.md [hunk 8]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+
+# Configure custom/insecure registry to search path in OCP 4.x
+
+At this moment, there seems to be no way to remove/modify or append any additional entries to unqualified-search-registries line in /etc/containers/registries.conf file as of now (OCP <=4.3).
+
+A workaround to modify the file is using a machineconfig that do the job that should be done by `image.config.openshift.io/cluster` (full control of the /etc/containers/{policy.json,registries.conf} content).
+
+Yo have to be aware that it could gets overwritten by editing `image.config.openshift.io/cluster` which will execute `99-master/worker-<uuid>-registries` and bring back the `unqualified-search-registries` again to the file, so you must avoit to use the `image.config.openshift.io/cluster` afeter you applied the machineconfig. 
+
+
```
Comment (e-minguez) [inline content/registries/_index.md]: The following snippet shows the content of the `machineconfig` object that modifies the `unqualified-search-registries` parameter:

File: content/registries/_index.md [hunk 9]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+        mode: 420
+        path: /etc/containers/policy.json
+    systemd: {}
+  fips: false
+  kernelArguments: null
+  osImageURL: ""
+```
+
+> NOTE: <MachineConfigPool> possible values are **worker** or **master**.
+> <base64_content> is the content you want to put into the config files.
```
Comment (e-minguez) [inline content/registries/_index.md]: NOTE: <MachineConfigPool> possible values are **worker** or **master**.
<base64_content> is the full content of the config files (including the `unqualified-search-registries` parameter).

In order to get the entire content of the file, you can connect to any of the hosts and extract the content (you can use `oc debug node/<mynode>` as cluster-admin user)

File: content/registries/_index.md [hunk 10]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+
+You must apply the MachineConfig (master, worker or both it depends of your needs)
```
Comment (e-minguez) [inline content/registries/_index.md]: Once the previous file is properly created with the proper file content and the roles to be applied (master, worker, etc.), the `MachineConfig` needs to be applied to the cluster as:

File: content/registries/_index.md [hunk 11]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+```
+oc create -f 99_<MachineConfigPool>_container_runtime.yaml
+```
```
Comment (e-minguez) [inline content/registries/_index.md]: Add a new line here

File: content/registries/_index.md [hunk 12]
```diff
@@ -13,3 +13,61 @@ weight: 23
...
+Then wait for clusteroperators to finish and nodes to be ready and then one can confirm the modification by executing the command below:
```
Comment (e-minguez) [inline content/registries/_index.md]: The modification will trigger a reboot on the hosts, so in order to wait for the nodes to be ready, you can use the command below:
